File: src/backend/Services/Reservations/OrangeCarRental.Reservations.Application/Services/PriceCalculationDto.cs
namespace SmartSolutionsLab.OrangeCarRental.Reservations.Application.Services;

/// <summary>
///     DTO for price calculation results from the Pricing API.
/// </summary>
public sealed record PriceCalculationDto(
    string CategoryCode,
    int TotalDays,
    decimal DailyRateNet,
    decimal DailyRateGross,
    decimal TotalPriceNet,
    decimal TotalPriceGross,
    decimal VatAmount,
    decimal VatRate,
    string Currency);

File: src/frontend/libs/data-access/src/lib/models/pricing/price-calculation.dto.ts
/**
 * Price calculation result matching backend PriceCalculationResult
 * All prices include German VAT (19%)
 */
import type { Currency, Price } from '../common';
import type { CategoryCode, DailyRate } from '../vehicle';
import type { TotalDays } from '../reservation';
import type { VatRate } from './vat-rate.type';

export type PriceCalculation = {
  readonly categoryCode: CategoryCode;
  readonly totalDays: TotalDays;
  readonly dailyRateNet: DailyRate;
  readonly dailyRateGross: DailyRate;
  readonly totalPriceNet: Price;
  readonly totalPriceGross: Price;
  readonly vatAmount: Price;
  readonly vatRate: VatRate;
  readonly currency: Currency;
};
